include/spi.h

```diff
@@ -141,6 +146,32 @@ struct spi_cs_control {
 	u32_t		delay;
 };
 
+/**
+ * @brief Slave events
+ */
+enum spi_slave_event {
+	/** When slave chip is deselected */
+	SPI_SLAVE_EVENT_DESELECTED	= 0,
+	/** When slave chip is selected */
+	SPI_SLAVE_EVENT_SELECTED,
+	/** When slave need to reply to master input */
+	SPI_SLAVE_EVENT_INPUT,
+	/** When an error occured, stopping the slave */
+	SPI_SLAVE_EVENT_ERROR,
+};
+
+struct spi_config;
+
+/**
+ * @typedef spi_slave_cb
+ * @brief Callback API for SPI slave events
+ *
+ * @param config Pointer to the current slave spi_config structure instance.
+ * @param event
+ */
+typedef int (*spi_slave_cb)(const struct spi_config *config,
```

Comment:
What is the purpose of the return value?

Reply:
I am not entirely decided on that, but here is the idea: let's say you get an input event, and in your slave code you check what the master has sent you so far. And it's full garbage: shall we notify to the driver so it can stop the transaction by returning an error code?